server/base/src/main/java/org/apache/accumulo/server/ServerOpts.java

```diff
@@ -27,6 +28,17 @@ public class ServerOpts extends ConfigOpts {
   @Parameter(names = {"-a", "--address"}, description = "address to bind to")
   private String address = null;
 
+  @Parameter(required = false, names = {"-g", "--group"},
+      description = "Optional group name that will be made available to the client (e.g. ScanServerSelector) "
+          + "and server (e.g. Balancers) plugins.  If not specified will be set to '"
+          + ServiceLockData.ServiceDescriptor.DEFAULT_GROUP_NAME
+          + "'. Assigning servers to groups support dedicating server resources for specific purposes, where supported.")
+  private String groupName = ServiceLockData.ServiceDescriptor.DEFAULT_GROUP_NAME;
```

Comment:
I've never been a fan of these command-line options separate from our config file settings. The `-a` was added as a fix, but the semantics of it have always been confusing, as in... is it the "bind address" or is it the "service advertisement" address? It has always been conflated as both, with a special case for `0.0.0.0` and `localhost`, but it should always have been two.

I'm mentioning this now, because I don't think we should continue the precedent of adding new command-line options. Instead, there should be a `tserver.group` attribute that can be put in the config file or specified on the command-line, like any other config file option can be, with `-o tserver.group=X`. (Similarly, we should have `tserver.address.listen` and `tserver.address.advertise` as separate properties that default to `0.0.0.0` and `AUTO` by default, but that's a separate issue. My main point here is that we shouldn't continue the previous precedent... we should do it via config).

Reply:
Does the current configuration mechanism support shell interpolation for configuration values? For example, could you put the following in `accumulo.properties`?
```
tserver.address.listen=$(hostname -i)`
```

If so, then this makes sense to me. If not, then I don't think anyone would want to put the bind address in a config file, as that would cause different `accumulo.properties` files on every host. At least by putting in on the command line you can use CLI commands to get the IP address and pass that in as the argument value. It even works in [k8s](https://github.com/dlmarion/accumulo-k8s/blob/main/accumulo-tserver.yaml#L23)

Reply:
We use commons-configuration2, which does support property interpolation, but not exactly like your example. You can do something like `tserver.address.listen=${env:HOSTNAME}` and in in your environment, you can do `HOSTNAME=$(hostname -i)` (for example, in `accumulo-env.sh` or `.bashrc`).

We also support setting any of our configuration properties on the command-line to override what's in the config file. So, instead of `-a $(hostname -i)`, you could do `-o tserver.address.listen=$(hostname -i)`, so you don't need to have the address hard-coded in the config file, and can set it dynamically on the command-line, even though it's a configuration file property. This feature was specifically added in 2.0 to support container deployments.

For the address, that would depend on us actually having a configuration property for the listen/bind address to replace `-a`, which we currently do not have (`tserver.address.listen` was merely an example). But, that's the direction I think we should go; for the purposes of this issue, I think the addition of the group should follow the direction of making it a configuration property, instead of the old precedent of `-a`. We have a complete configuration mechanism with a lot of flexibility already, so we shouldn't need to add one-off command-line options outside that configuration mechanism, like `-g` (and we should eventually get rid of the other one-off command-line options like `-a`, but that's for another ticket).